pre-docs/applying_manifest_blocks.md

```diff
@@ -0,0 +1,189 @@
+   mod 'profiles', local: true
+   ```
+1. Add the following code to the manifest `profiles/plans/iis_install.pp`
+   ```
+   plan profiles::iis_install(
+     TargetSpec $nodes,
+     String $site_content = 'hello!',
+   ) {
+     $nodes.apply_prep
+
+     return apply($nodes, '_catch_errors' => true) {
+       $iis_features = ['Web-WebServer','Web-Scripting-Tools']
+
+       iis_feature { $iis_features:
+         ensure => 'present',
+       }
+
+       # Delete the default website to prevent a port binding conflict.
+       iis_site {'Default Web Site':
+         ensure  => absent,
+         require => Iis_feature['Web-WebServer'],
+       }
+
+       iis_site { 'minimal':
+         ensure          => 'started',
+         physicalpath    => 'c:\\inetpub\\minimal',
+         applicationpool => 'DefaultAppPool',
+         require         => [
+           File['minimal'],
+           Iis_site['Default Web Site']
+         ],
+       }
+
+       file { 'minimal':
+         ensure => 'directory',
+         path   => 'c:\\inetpub\\minimal',
+       }
+
+       file { 'content':
+         ensure  => 'file',
+         path    => 'c:\\inetpub\\minimal\\index.html',
+         content => $site_content,
+       }
+     }
+   }
+   ```
+1. Run `bolt plan run profiles::iis_install --nodes <NODE NAME> --transport winrm`
+1. From a web browser, navigate to `<NODE NAME>` and you should see a page saying `hello!`
+
+### Explanation
+
+The `apply_prep` function sets up the remote node by installing our puppet-agent package from puppet.com using a `puppet_agent::install` task (from [the puppet_agent module](https://github.com/puppetlabs/puppetlabs-puppet_agent)) if Puppet is not detected on a target. It also copies over custom facts from Bolt's modulepath and executes `facter` on the targets.
+
+Behind the scenes, Bolt compiles the manifest block (the section wrapped in curly braces following `apply`) into a catalog using (from highest to lowest precedence)
+- `Facts` gathered from the targets or set in your inventory
+- Local variables in the plan, such as `$site_content`
+- `Vars` set in your inventory
+
+It generates all the variables a `puppet apply` would include, so code can be reused between Bolt and Puppet. It then copies custom module content from Bolt's modulepath to the target and applies the catalog using Puppet.
+
+Almost all features of the Pupet language - classes, custom resource types, functions - can be used, with exceptions noted in [caveats](#caveats). The nginx example above could make use of the [puppet-nginx](https://github.com/voxpupuli/puppet-nginx) module for more complex deployments.
+
+When the catalog compiles and is executed successfully on all targets, apply returns the reports generated by applying the catalog on each node.
+
+## Using Hiera Data
+
+Hiera is a tool for looking up data based on properties of a particular target. It allows you to separate configuration from context-specific data, such as facts or name of a target.
```

Comment:
This is a great 2 sentence summation of hiera!

Reply:
I think the very end is a little fuzzy, going to switch `such as facts or name of a target` to `where-as context may be facts or name of a target`.